<launch>

	<arg name="model_rosi" default="$(find rosi_defy)/urdf/rosi.urdf" />
	<param name="robot_description" command="$(find xacro)/xacro $(arg model_rosi)" />
	<node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" />
    <group ns="rtabmap">

        <node pkg="rtabmap_ros" type="rgbd_odometry" name="rgbd_odometry" output="screen">
            <remap from="rgb/image"       to="/sensor/kinect_rgb"/>
            <remap from="depth/image"     to="/sensor/kinect_depth"/>
            <remap from="rgb/camera_info" to="/sensor/kinect_info"/>
            <param name="odom_frame_id" type="string" value="odom" />	  
            <param name="Odom/Strategy" type="string" value="0"/>      <!-- 0=Frame-to-Map, 1=Frame-to-KeyFrame -->
            <param name="Odom/ResetCountdown" type="string" value="15"/> 
            <param name="Odom/GuessSmoothingDelay" type="string" value="0"/>

            <param name="frame_id" type="string" value="base_link"/>
            <param name="queue_size" type="int" value="10"/>
            <param name="wait_for_transform" type="bool" value="true"/>
            <param name="ground_truth_frame_id" type="string" value="world"/>
            <param name="ground_truth_base_frame_id" type="string" value="kinect_gt"/>
        </node>        
        <node name="rtabmap" pkg="rtabmap_ros" type="rtabmap" output="screen" args="--delete_db_on_start">
            <param name="frame_id" type="string" value="base_link"/>           
            <param name="subscribe_scan_cloud" type="bool" value="true"/>
            <remap from="scan_cloud" to="/sensor/velodyne"/>
            <remap from="rgb/image" to="/sensor/kinect_rgb"/>
            <remap from="depth/image" to="/sensor/kinect_depth"/>
            <remap from="rgb/camera_info" to="/sensor/kinect_info"/>
            <remap from="odom" to="/rtabmap/odom" />
        </node>
        <node pkg="rtabmap_ros" type="rtabmapviz" name="rtabmapviz" args="-d $(find rtabmap_ros)/launch/config/rgbd_gui.ini" output="screen">
            <param name="frame_id" type="string" value="base_link"/>
            <param name="subscribe_rgb" type="bool" value="false"/>            
            <param name="subscribe_depth" type="bool" value="false"/>
            <param name="subscribe_scan_cloud" type="bool" value="true"/>
            <remap from="scan_cloud" to="/sensor/velodyne"/>
        </node>

    </group>
</launch>
